### PR TITLE
fix: chart publication logic

### LINKED ISSUE #580: helm chart does not get published
**Describe the bug:** (a clear and concise description of what the bug is)

When a change to the version in `chart/otomi/Chart.yaml` gets pushed to master this does not always trigger a new chart release.

Reason:

The logic that checks for diff changes in `chart/otomi/Chart.yaml` is only taking into account the previous commit, but the pipeline containing the version change may have been skipped with `[ci skip]`.

Solution suggestions:

1. Retrieve the latest released version with:
```bash
helm repo update
chart_version=$(helm search repo otomi | sed -n 2p | awk '{print $2}')
```

and compare with current version in package json

**To Reproduce**

1. create a commit with `[ci skip]` in the subject but don't push
2. change `version` in `chart/otomi/Chart.yaml` and commit
3. now do a git push and observe the pipeline step `Create and publish otomi chart release` gets skipped

**Expected behavior:** (a clear and concise description of what you expected to happen)

Chart to always be published.

**Screenshots:** (if applicable, add screenshots to help explain your problem)

**Cluster(s):**

- Cloud: [e.g. aws]
- Cluster: [e.g. demo]

**Versions:** (seen in console's `/settings`)

- Core Version [e.g. 0.12.34]
- Console Version: [e.g. aws]
- API version: [e.g. 0.4.56]

**Desktop:** (if relevant)

**Smartphone:** (if relevant)

**Additional context:** (add any other context about the problem here)

### PR DESCRIPTION
**Pull request by Morriz**
fixes #580

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -74,21 +74,45 @@ jobs:
...
+      - name: Check for changes
+        id: prereq
         run: |
           # Release new chart only if its version changes
-          [ ! $(git show ${{ github.sha }} chart/otomi/Chart.yaml | grep '^+version:.*$') ] && echo "No chart changes detected. Skipping." && exit
-          echo ::set-output name=has_chart_changes::true
+          helm repo add otomi https://otomi.io/otomi-core
+          helm repo update
+          this_version=v$(cat chart/otomi/Chart.yaml | yq r - 'version')
+          chart_version=$(helm search repo otomi | sed -n 2p | awk '{print $2}')
```
Comment (j-zimnowoda) [inline .github/workflows/main.yml:105]: Can we read chart version in more explicit way
```
helm search repo otomi -o json | jq  -r '.[-1].version'
```
Please be aware that in both cases we assume that there is only one chart in chart repo

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Looking good, just one remark

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -74,21 +74,45 @@ jobs:
...
+          SLACK_USERNAME: RedKubesBot
+        if: always() # failure
+
+  chart-release:
+    if: "contains(github.event.head_commit.message, 'chore(release)')"
+    needs: build-test-push-release
+    runs-on: ubuntu-latest
+    container:
+      image: otomi/tools:v1.4.19
+      options: --user 0
```
Comment (j-zimnowoda) [inline .github/workflows/main.yml:94]: is it possible with non-root user?

File: .github/workflows/main.yml (hunk 1 above)
Comment (Morriz) [reply .github/workflows/main.yml:105]: unfortunately not, as that would pull the oldest version, not the latest...can you give me a selector that gets the latest version then?

(I tried with `helm search repo jetstack/cert-manager -o json | jq  -r '.[-1].version'`)

File: .github/workflows/main.yml (hunk 2 above)
Comment (Morriz) [reply .github/workflows/main.yml:94]: no, as the pipeline runner needs root access to mount certain things...

File: .github/workflows/main.yml (hunk 1 above)
Comment (Morriz) [reply .github/workflows/main.yml]: solved with latest commit

**Pull request merged**